js/ui/date_box/ui.date_box.base.js

```diff
@@ -732,8 +727,12 @@ const DateBox = DropDownEditor.inherit({
             this._saveValueChangeEvent(dxEvent);
         }
 
-        if(!isValueChanged && this._isTextChanged(value)) {
-            this._updateValue(value);
+        if(!isValueChanged) {
+            if(this._isTextChanged(value)) {
+                this._updateValue(value);
+            } else if(this.option('text') === '') {
+                this._applyCustomValidation(value);
```

Comment:
Can we remove this line?

Reply:
No. Let's imagine the scenario with custom validation rule 'required':
1) Type some letters -> internalValidation is failed (value and text do not change)
2) Clear the input -> internalValidation is ok; customValidation is not fired because text has not been changed.
--> validation is ok, but customValidation should be fired and fail

Reply:
Ok.